lib/actions-oclif/envs.ts

```diff
@@ -93,28 +129,50 @@ export default class EnvsCmd extends Command {
 		(for example, in case the current user has been removed from the application
 		by its owner).
 `;
-	public static examples = [
-		'$ balena envs --application MyApp',
-		'$ balena envs --application MyApp --all --json',
-		'$ balena envs --application MyApp --service MyService',
-		'$ balena envs --application MyApp --all --service MyService',
-		'$ balena envs --application MyApp --config',
-		'$ balena envs --device 7cf02a6',
-		'$ balena envs --device 7cf02a6 --all --json',
-		'$ balena envs --device 7cf02a6 --config --all --json',
-		'$ balena envs --device 7cf02a6 --all --service MyService',
-	];
+	public static examples = isV12()
+		? [
+				'$ balena envs --application MyApp',
+				'$ balena envs --application MyApp --json',
+				'$ balena envs --application MyApp --service MyService',
+				'$ balena envs --application MyApp --service MyService',
+				'$ balena envs --application MyApp --config',
+				'$ balena envs --device 7cf02a6',
+				'$ balena envs --device 7cf02a6 --json',
+				'$ balena envs --device 7cf02a6 --config --json',
+				'$ balena envs --device 7cf02a6 --service MyService',
+		  ]
+		: [
+				'$ balena envs --application MyApp',
+				'$ balena envs --application MyApp --all --json',
+				'$ balena envs --application MyApp --service MyService',
+				'$ balena envs --application MyApp --all --service MyService',
+				'$ balena envs --application MyApp --config',
+				'$ balena envs --device 7cf02a6',
+				'$ balena envs --device 7cf02a6 --all --json',
+				'$ balena envs --device 7cf02a6 --config --all --json',
+				'$ balena envs --device 7cf02a6 --all --service MyService',
+		  ];
 
 	public static usage = (
 		'envs ' + new CommandHelp({ args: EnvsCmd.args }).defaultUsage()
 	).trim();
 
 	public static flags: flags.Input<FlagsDef> = {
-		all: flags.boolean({
-			description: stripIndent`
+		...(isV12()
+			? {
+					all: flags.boolean({
+						description: stripIndent`
+				No-op since balena CLI v12.0.0. See "balena help envs".`,
```

Comment:
I think _"See balena help envs"_ could be removed, as `balena help envs` doesn't clarify much regarding the `--all` option. I guess you got inspiration for it from my previous PR (`--nogitignore`), but in the case of that PR, the `balena help push` output included a paragraph that actually mentioned `.gitignore` files and explained that _"Previous balena CLI releases (before v12.0.0) also took '.gitignore' files into account, but this is no longer the case."_